gcc/config/riscv/riscv.c

```diff
@@ -3590,6 +3656,14 @@ riscv_expand_epilogue (bool sibcall_p)
   bool need_barrier_p = (get_frame_size ()
 			 + cfun->machine->frame.arg_pointer_offset) != 0;
 
+  if (cfun->machine->naked_p)
+    {
+      if (!sibcall_p)
```

Comment:
This is a little confusing at first glance, but since a naked function can't be a sibcall, this is just unconditionally calling emit_jump_insn.  It could maybe be written that way.